nodes/src/nodes/agent_deepagent/deepagent.py

```
@@ -768,45 +925,127 @@ def _parse_tool_call_envelope(raw: str) -> Any:
+
+    Args:
+        base: The primary system prompt string, or ``None``/empty for the fallback.
+        instructions: Optional list of instruction lines to append.
+        fallback: Default prompt text used when *base* is empty after stripping.
+
+    Returns:
+        A single system-prompt string ready to hand to ``create_deep_agent``.
+    """
+    prompt = (base or '').strip() or fallback
+    for inst in instructions or []:
+        inst = inst.strip()
```

Ruff (0.15.9)

[warning] `for` loop variable `inst` overwritten by assignment target

(PLW2901)